### PR TITLE
OP-545: Twig templates for blocks and pages

### PR DESCRIPTION
**Pull request by jkindly**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes

### CODE REVIEW

File: src/Resources/config/services/provider.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <defaults public="true" />
```
Comment (senghe) [inline src/Resources/config/services/provider.xml:3]: Why it's public?

**Review by senghe — APPROVED** [review]
It's good. There're two short changes, so I can approve in advance. When you change them, you can go with merge.

File: src/Controller/PageController.php [hunk 2]
```diff
@@ -22,6 +25,42 @@ final class PageController extends ResourceController
...
+    public function showAction(Request $request): Response
+    {
+        $configuration = $this->getRequestConfiguration($request);
+
+        $this->isGrantedOr403($configuration, ResourceActions::SHOW);
+
+        $slug = $request->attributes->get('slug');
+
+        /** @var PageRepositoryInterface $pageRepository */
+        $pageRepository = $this->get('sylius_cms.repository.page');
```
[the remark this answers is not in the document]
Comment (jkindly) [reply src/Controller/PageController.php:39]: Me too, but the problem here is that PageController extends ResourceController and constructor will looks like:
`    public function __construct(MetadataInterface $metadata, RequestConfigurationFactoryInterface $requestConfigurationFactory, ?ViewHandlerInterface $viewHandler, RepositoryInterface $repository, FactoryInterface $factory, NewResourceFactoryInterface $newResourceFactory, ObjectManager $manager, SingleResourceProviderInterface $singleResourceProvider, ResourcesCollectionProviderInterface $resourcesFinder, ResourceFormFactoryInterface $resourceFormFactory, RedirectHandlerInterface $redirectHandler, FlashHelperInterface $flashHelper, AuthorizationCheckerInterface $authorizationChecker, EventDispatcherInterface $eventDispatcher, ?StateMachineInterface $stateMachine, ResourceUpdateHandlerInterface $resourceUpdateHandler, ResourceDeleteHandlerInterface $resourceDeleteHandler)
    {
        parent::__construct($metadata, $requestConfigurationFactory, $viewHandler, $repository, $factory, $newResourceFactory, $manager, $singleResourceProvider, $resourcesFinder, $resourceFormFactory, $redirectHandler, $flashHelper, $authorizationChecker, $eventDispatcher, $stateMachine, $resourceUpdateHandler, $resourceDeleteHandler);
    }`
    
    So, I can't use DI here.

File: src/Resources/config/services/provider.xml (hunk 1 above)
Comment (jkindly) [reply src/Resources/config/services/provider.xml:3]: Every service in services dir is public, so I went with the project flow. Should I remove it?

Comment (senghe) [reply src/Resources/config/services/provider.xml:3]: It shouldn't be public, as public services aren't good for containers performance. But please leave it as is.

**Pull request merged**